src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/Hop.tsx

```diff
@@ -9,72 +9,82 @@ import {
   Flex,
   HStack,
   IconButton,
-  Spacer,
-  Step,
-  StepDescription,
-  StepIndicator,
   Stepper,
-  StepSeparator,
-  StepTitle,
   useColorModeValue,
 } from '@chakra-ui/react'
 import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { getDefaultSlippageDecimalPercentageForSwapper } from 'constants/constants'
 import prettyMilliseconds from 'pretty-ms'
-import { useMemo } from 'react'
+import { useMemo, useState } from 'react'
 import { FaAdjust, FaGasPump, FaProcedures } from 'react-icons/fa'
 import { Amount } from 'components/Amount/Amount'
-import type { StepperStep } from 'components/MultiHopTrade/types'
 import { RawText } from 'components/Text'
+import type { SwapperName, TradeQuoteStep } from 'lib/swapper/types'
+import { assertUnreachable } from 'lib/utils'
+import {
+  selectHopTotalNetworkFeeFiatPrecision,
+  selectHopTotalProtocolFeesFiatPrecision,
+  selectInitialApprovalRequirements,
+  selectIsActiveQuoteMultiHop,
+  selectTradeExecutionState,
+} from 'state/slices/tradeQuoteSlice/selectors'
+import { HopExecutionState } from 'state/slices/tradeQuoteSlice/types'
+import { store, useAppSelector } from 'state/store'
 
+import { TradeType } from '../types'
+import { getHopExecutionState } from '../utils/getHopExecutionState'
+import { ApprovalStep } from './ApprovalStep'
+import { AssetSummaryStep } from './AssetSummaryStep'
+import { DonationStep } from './DonationStep'
+import { HopTransactionStep } from './HopTransactionStep'
 import { JuicyGreenCheck } from './JuicyGreenCheck'
 import { TimeRemaining } from './TimeRemaining'
 
 const cardBorderRadius = { base: 'xl' }
 
 export const Hop = ({
-  steps,
-  activeStep,
-  slippageDecimalPercentage,
-  networkFeeFiatPrecision,
-  protocolFeeFiatPrecision,
-  title,
+  swapperName,
+  tradeQuoteStep,
   hopIndex,
-  txStatus,
   isOpen,
-  estimatedExecutionTimeMs,
   onToggleIsOpen,
 }: {
-  steps: StepperStep[]
-  activeStep: number
-  slippageDecimalPercentage: string
-  networkFeeFiatPrecision: string
-  protocolFeeFiatPrecision: string
-  title: string
+  swapperName: SwapperName
+  tradeQuoteStep: TradeQuoteStep
   hopIndex: number
-  txStatus?: TxStatus
   isOpen: boolean
-  estimatedExecutionTimeMs?: number
   onToggleIsOpen: () => void
 }) => {
   const backgroundColor = useColorModeValue('gray.100', 'gray.750')
   const borderColor = useColorModeValue('gray.50', 'gray.650')
-
   const chevronUpIcon = useMemo(() => <ChevronUpIcon boxSize='16px' />, [])
   const chevronDownIcon = useMemo(() => <ChevronDownIcon boxSize='16px' />, [])
+  const networkFeeFiatPrecision = selectHopTotalNetworkFeeFiatPrecision(store.getState(), hopIndex)
+  const protocolFeeFiatPrecision = selectHopTotalProtocolFeesFiatPrecision(
+    store.getState(),
+    hopIndex,
+  )
```

Comment:
Is this expected to use non-reactive, pure selector methods?

Reply:
this was in error, resolved with 53df4723c4d5a51d74833f2fce422bacf01a889e

Reply:
this was in error, resolved with 53df4723c4d5a51d74833f2fce422bacf01a889e